src/sim/syscall_emul.hh

```diff
@@ -677,6 +678,73 @@ copyOutStatfsBuf(TgtStatPtr tgt, HostStatPtr host)
 #endif
 }
 
+template <typename OS, typename TgtStatPtr, typename HostStatPtr>
+void
+copyOutStatxBuf(TgtStatPtr tgt, HostStatPtr host, bool fakeTTY = false)
+{
+    constexpr ByteOrder bo = OS::byteOrder;
+
+    if (fakeTTY) {
+        tgt->stx_dev_major = MAJOR(0xA);
+        tgt->stx_dev_minor = MINOR(0xA);
+    } else {
+        tgt->stx_dev_major = host->stx_dev_major;
+        tgt->stx_dev_minor = host->stx_dev_minor;
+    }
+    tgt->stx_dev_major = htog(tgt->stx_dev_major, bo);
+    tgt->stx_dev_minor = htog(tgt->stx_dev_minor, bo);
+    tgt->stx_ino = host->stx_ino;
+    tgt->stx_ino = htog(tgt->stx_ino, bo);
```

Comment:
I checked other copyOut helpers and I see why you are doing this in two assignments (for consistency).
I honestly believe we should break our past habit and write this in one :) :

    tgt->stx_ino = htog(host->stx_ino, bo);


Same applies to the rest of the file

Reply:
I think writing it as one statement in that way could lead to buggy results if the host and target have fields of different sizes. E.g., say the host has a 64-bit stx_ino field but the target has a 32-bit stx_ino field. When writing it as two statements, it (1) narrows the integer to 32 bits, then (2) possibly reverses the byte order.
But if writing it as one statement as above, those operations happen in the reverse order: (1) reverse the byte order, then (2) narrow integer to 32 bits.

I think this is a pathological case, since statx is only implemented on Linux currently I believe, so we'd expect the statx member sizes to always match from host to target. Let me know if the above isn't a concern and I can rewrite all 2-statement assignments into 1-statement assignments.

Reply:
Yeah you are absolutely right, I missed this case